### ISSUE TITLE
ActiveSupport::Cache::FileStore#cleanup does not remove expired entries from cache

### ISSUE THREAD

**Issue by erichmachado** [rails/rails#30788]
### Steps to reproduce
```ruby
# frozen_string_literal: true

begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "activesupport", "5.1.0"
end

require "active_support"
require "minitest/autorun"

# Ensure backward compatibility with Minitest 4
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)

class BugTest < Minitest::Test
  def setup
    Dir.mkdir(cache_dir) unless File.exist?(cache_dir)
    @cache = ActiveSupport::Cache.lookup_store(:file_store, cache_dir, expires_in: 60)
  end

  def teardown
    FileUtils.rm_r(cache_dir)
  rescue Errno::ENOENT
  end

  def cache_dir
    File.join(Dir.pwd, "tmp_cache")
  end

  def test_cleanup_removes_all_expired_entries
    time = Time.now
    @cache.write("foo", "bar", expires_in: 10)
    @cache.write("baz", "qux")
    @cache.write("quux", "corge", expires_in: 20)
    Time.stub(:now, time + 15) do
      @cache.cleanup
      refute @cache.exist?("foo")
      assert @cache.exist?("baz")
      assert @cache.exist?("quux")
      assert_equal 2, Dir.glob(File.join(cache_dir, "**")).size
    end
  end
end
```

### Expected behavior
The number of entries in the cache store directory should equal the number of non-expired entries after the cleanup call.

### Actual behavior
The cleanup call is not removing the expired entries from the cache directory. The test was returning a false positive since it was relying exclusively on the `ActiveSupport::Cache::Store#exist?` implementation, which returns `false` if the entry still exists in the storage but is expired.

### System configuration
**Rails version**: 5.1.0 (but also affects earlier versions, checked also with 5.0.0)

**Ruby version**: 2.4.2

**Comment by y-yagi** [comment]
This fixed by #30789.

**Comment by bnymn** [comment]
This PR did not go to Rails 4.2.10?

**Comment by erichmachado** [comment]
Hi @bnymn, I believe it was merged only on 5.2.0. I think 4.2 is not receiving updates anymore.

**Comment by sikachu** [comment]
Yep, the first version that this commit was included is v5.2.0.beta1. At this point 4.2.x only receive patch for severe security issues.
